emcc.py

```diff
@@ -2275,7 +2275,7 @@ def do_binaryen(final, target, asm_target, options, memfile, wasm_binary_target,
     log_time('asm2wasm')
   if shared.Settings.BINARYEN_PASSES:
     shutil.move(wasm_binary_target, wasm_binary_target + '.pre')
-    cmd = [os.path.join(binaryen_bin, 'wasm-opt'), wasm_binary_target + '.pre', '-o', wasm_binary_target] + map(lambda p: '--' + p, shared.Settings.BINARYEN_PASSES.split(','))
+    cmd = [os.path.join(binaryen_bin, 'wasm-opt'), wasm_binary_target + '.pre', '-o', wasm_binary_target] + map(lambda p: ('--' + p) if p[0] != '-' else p, shared.Settings.BINARYEN_PASSES.split(','))
```

Comment:
That `map`/`lambda` is getting pretty complex. Might be worth pulling it to an intermediate variable on the line above to help break it up?
Something like `passes = map(lambda p: ...`